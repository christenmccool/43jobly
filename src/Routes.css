.Routes {
  width: 65%;
  margin: 40px auto;
}
