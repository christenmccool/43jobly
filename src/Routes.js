import React, {useContext} from 'react';
import {Switch, Route, Redirect} from 'react-router-dom';
import Home from './Home';
import CompanyPage from './CompanyPage';
import CompanyDetail from './CompanyDetail';
import JobPage from './JobPage';
import UserApps from './UserApps';
import LoginForm from './LoginForm';
import SignupForm from './SignupForm';
import ProfileForm from './ProfileForm';
import NotFound from './NotFound';
import UserContext from './UserContext';
import './Routes.css';

/** Routes component for Jobly app
 * Redirects to NotFound component if path not found
 */

const Routes = ({signup, login, editProfile, apply}) => {
  const user = useContext(UserContext);

  if (user) {
    return (
      <div className="Routes">
        <Switch>
          <Route exact path="/">
            <Home />
          </Route>
          <Route exact path="/companies">
            <CompanyPage />
          </Route>
          <Route exact path="/companies/:handle">
            <CompanyDetail apply={apply}/>
          </Route>
          <Route exact path="/jobs">
            <JobPage apply={apply}/>
          </Route>
          <Route exact path="/applications">
            <UserApps />
          </Route>
          <Route exact path="/login">
            <Redirect to="/"></Redirect>
          </Route>
          <Route exact path="/signup">
            <SignupForm signup={signup}/>
          </Route>
          <Route exact path="/profile">
            <ProfileForm editProfile={editProfile}/>
          </Route>
          <Route exact path="/404">
            <NotFound />
          </Route>
          <Redirect to="/404"></Redirect>
        </Switch>
      </div>
    )
    } else {
      return (
        <div className="Routes">
          <Switch>
            <Route exact path="/">
              <Home />
            </Route>
            <Route exact path="/companies">
              <Redirect to="/login"></Redirect>
            </Route>
            <Route exact path="/companies/:handle">
              <Redirect to="/login"></Redirect>
            </Route>
            <Route exact path="/jobs">
              <Redirect to="/login"></Redirect>
            </Route>
            <Route exact path="/myjobs">
              <Redirect to="/login"></Redirect>
            </Route>
            <Route exact path="/login">
              <LoginForm login={login}/>
            </Route>
            <Route exact path="/signup">
              <SignupForm signup={signup}/>
            </Route>
            <Route exact path="/profile">
              <Redirect to="/login"></Redirect>
            </Route>
            <Redirect to="/"></Redirect>
          </Switch>
        </div>
      )
    }
}

export default Routes;