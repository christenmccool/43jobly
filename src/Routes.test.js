import { render, asFragment, fireEvent, waitFor } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Routes from './Routes';

test('if renders without crashing', () => {
  render(<MemoryRouter><Routes /></MemoryRouter>);
});

test('if matches snapshot', () => {
  const {asFragment} = render(<MemoryRouter><Routes /></MemoryRouter>);
  expect(asFragment()).toMatchSnapshot();
});