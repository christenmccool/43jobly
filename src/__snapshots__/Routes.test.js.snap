// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Routes"
  >
    <div
      class="Home"
    >
      <h1>
        Jobly
      </h1>
      <p>
        All the jobs in one convenient place.
      </p>
      <div>
        <a
          href="/login"
        >
          Log in
        </a>
        <a
          href="/signup"
        >
          Sign up
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
